bay/docker/images.py

```diff
@@ -187,6 +190,15 @@ def _pull(self, app, task, remote_name, image_tag, tries=0):
             # we should always have Docker images uploaded on ECR, but if for some
             # reason the image can't be found, we raise an error and it will get built
             # instead
+            if 'BAY_NO_REGISTRY' not in os.environ:
+                # Sometimes docker python client fails with NotFound error while trying to pull
+                #   an image without having valid credentials.
+                # For the user, it means that all the static images would be built locally. It's a long process.
+                # We want to fail fast in that case and give the user the chance to login to the registry.
+                click.echo(RED("Cannot pull image {}:{}.".format(remote_name, image_tag)))
+                click.secho("To fix this, please run `bay registry login`.", bold=True)
```

Comment:
Are not we checking the access to registry before pull ?

Reply:
We always do a `docker login` and we checked that the login was successful.
So it should not happen, but a few users have reported getting "not found" errors and it was solved by running `bay registry login`.

Reply:
ok, Thank you!